pyface/ui/qt4/progress_dialog.py

```diff
@@ -97,6 +97,20 @@ def close(self):
 
         super(ProgressDialog, self).close()
 
+    # -------------------------------------------------------------------------
+    # 'IWidget' interface.
+    # -------------------------------------------------------------------------
+
+    def destroy(self):
+        if self.control is not None:
+            buttons = self.control.findChild(QtGui.QDialogButtonBox)
```

Comment:
I agree that this is smelly, could we hold a reference to the button created by this class and use that, and then remove the reference in `destroy`.